packages/nitro-protocol/hardhat.config.ts

```diff
@@ -5,36 +5,49 @@
 import 'hardhat-deploy';
 import 'hardhat-deploy-ethers';
 
-const infuraToken = process.env.INFURA_TOKEN;
-const rinkebyDeployerPK = process.env.RINKEBY_DEPLOYER_PK;
-if (!infuraToken) {
-  throw new Error(`Invalid infura token ${infuraToken}`);
-}
-
-if (!rinkebyDeployerPK) {
-  throw new Error(`Invalid deploy private key ${rinkebyDeployerPK}`);
-}
-
-module.exports = {
-  solidity: '0.6.12',
-  namedAccounts: {
-    deployer: {
-      // Address used for the initial deploy of the contracts.
-      // Ask @kerzhner for the private key if you would like to deploy from the same address
-      // Otherwise uncomment the line below
-      //rinkeby: 0,
-      rinkeby: '0x87612aAD373586A38062c29F833A2AbC72038591',
+const config = {
+  solidity: {
+    version: '0.6.12',
+    settings: {
+      outputSelection: {
+        '*': {
+          '*': [
+            'evm.bytecode.object',
+            'evm.deployedBytecode.object',
+            'abi',
+            'evm.bytecode.sourceMap',
+            'evm.deployedBytecode.sourceMap',
+          ],
+          '': ['ast'],
+        },
+      },
     },
   },
-  networks: {
-    rinkeby: {
-      url: 'https://rinkeby.infura.io/v3/' + infuraToken,
-      accounts: [rinkebyDeployerPK],
-    },
+  namedAccounts: {
+    deployer: {},
   },
+  networks: {},
   paths: {
     sources: 'contracts',
     deploy: 'hardhat-deploy',
     deployments: 'hardhat-deployments',
   },
 };
+
+const infuraToken = process.env.INFURA_TOKEN;
+const rinkebyDeployerPK = process.env.RINKEBY_DEPLOYER_PK;
+if (infuraToken && rinkebyDeployerPK) {
```

Comment:
What happens if `yarn hardhat deploy --network rinkeby` is run without these env vars defined? An error, or a silent failure?

Reply:
Good point. In 39116a1 I have restored the previous behaviour by making the current config file extend off of a leaner base config.